.github/workflows/ci.yml

```diff
@@ -130,16 +142,31 @@ jobs:
             cxxstd: "03,11,14,17,2a"
             os: macos-10.15
 
-    runs-on: ${{matrix.os}}
+    runs-on: ${{ matrix.os }}
+    container: ${{ matrix.container }}
 
     steps:
+      - name: Check if running in container
+        if: matrix.container != ''
+        run: echo "GHA_CONTAINER=${{ matrix.container }}" >> $GITHUB_ENV
+      - name: If running in container, upgrade packages
+        if: matrix.container != ''
+        run: |
+            apt-get -o Acquire::Retries=3 update && DEBIAN_FRONTEND=noninteractive apt-get -y install tzdata && apt-get -o Acquire::Retries=3 install -y sudo software-properties-common wget curl apt-transport-https make apt-file sudo unzip libssl-dev build-essential autotools-dev autoconf automake g++ libc++-helpers python ruby cpio gcc-multilib g++-multilib pkgconf python3 ccache libpython-dev
```

Comment:
Is `sudo` missing here?

Reply:
Is sudo missing here? Not on that line, no. The sudo package isn't pre-installed on all base docker images. It will be in the next step. After it's installed, then it can be used.